modules/apps/commerce/commerce-order-content-web/src/main/resources/META-INF/resources/js/returnableOrderItemsPropsTransformer.js

```diff
@@ -3,112 +3,13 @@
  * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
  */
 
-import {CommerceServiceProvider} from 'commerce-frontend-js';
-import {createPortletURL, openToast} from 'frontend-js-web';
-
-function returnableOrderItemsPropsTransformer({
-	additionalProps: context,
-	...props
-}) {
+function returnableOrderItemsPropsTransformer({...props}) {
 	return {
 		...props,
-		onBulkActionItemClick: async ({
-			selectedData: {items: commerceOrderItems},
-		}) => {
-			const CommerceReturnResource = CommerceServiceProvider.ReturnAPI();
-
-			const {
-				accountEntryId,
-				channelGroupId,
-				channelId,
-				channelName,
-				commerceOrderId,
-				commerceReturnId,
-				redirect,
-			} = context;
-
-			const commerceReturn = {
-				channelGroupId: parseInt(channelGroupId, 10),
-				channelId: parseInt(channelId, 10),
-				channelName,
-				commerceReturnToCommerceReturnItems: commerceOrderItems.map(
-					({
-						id: commerceOrderItemId,
-						price: {finalPrice: amount},
-						quantity,
-					}) => ({
-						amount,
-						quantity,
-						r_accountToCommerceReturnItems_accountEntryId: parseInt(
-							accountEntryId,
-							10
-						),
-						r_commerceOrderItemToCommerceReturnItems_commerceOrderItemId:
-							commerceOrderItemId,
-					})
-				),
-				r_accountToCommerceReturns_accountEntryId: parseInt(
-					accountEntryId,
-					10
-				),
-				r_commerceOrderToCommerceReturns_commerceOrderId: parseInt(
-					commerceOrderId,
-					10
-				),
-			};
-
-			if (parseInt(commerceReturnId, 10)) {
-				return CommerceReturnResource.updateItemById(
-					commerceReturnId,
-					commerceReturn
-				)
-					.then((response) => {
-						window.top.location.href = createPortletURL(redirect, {
-							commerceReturnId: response.id,
-						});
-
-						openToast({
-							message: Liferay.Language.get(
-								'your-request-completed-successfully'
-							),
-							type: 'success',
-						});
-					})
-					.catch((error) => {
-						openToast({
-							message:
-								error.message ||
-								Liferay.Language.get(
-									'an-unexpected-error-occurred'
-								),
-							type: 'danger',
-						});
-					});
-			}
-
-			return CommerceReturnResource.createItem(commerceReturn)
-				.then((response) => {
-					window.top.location.href = createPortletURL(redirect, {
-						commerceReturnId: response.id,
-					});
-
-					openToast({
-						message: Liferay.Language.get(
-							'your-request-completed-successfully'
-						),
-						type: 'success',
-					});
-				})
-				.catch((error) => {
-					openToast({
-						message:
-							error.message ||
-							Liferay.Language.get(
-								'an-unexpected-error-occurred'
-							),
-						type: 'danger',
-					});
-				});
+		onItemSelected: (selectedItems) => {
+			window.top.Liferay.fire('selected-returnable-items', {
```

Comment:
can we put the name of the event in a constant? (because I think we will use it also somewhere with the .on(..) )

Reply:
Hey @smottal if I remember correctly Brian is against constants, can you tell me if there are other points where constants are defined for events?